File: src/main/java/com/codethen/BotsRunner.java
package com.codethen;

import com.codethen.telegram.RegisterBots;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
public class BotsRunner implements CommandLineRunner {

    private final String connectionString;
    private final String databaseName;
    private final String lanxatToken;
    private final String lanxatName;

    public BotsRunner(
            @Value("${mongo.url}") String connectionString,
            @Value("${mongo.database}") String databaseName,
            @Value("${telegram.lanxat.token}") String lanxatToken,
            @Value("${telegram.lanxat.name}") String lanxatName) {
        this.connectionString = connectionString;
        this.databaseName = databaseName;
        this.lanxatToken = lanxatToken;
        this.lanxatName = lanxatName;
    }

    @Override
    public void run(String... args) throws Exception {
        RegisterBots.registerBots(connectionString, databaseName, lanxatName, lanxatToken);
    }
}

File: src/main/kotlin/com/codethen/BotsRunner.kt
package com.codethen

import com.codethen.telegram.RegisterBots
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

/**
 * Takes values from application.properties and register bots.
 */
@Component
class BotsRunner(
        @Value("\${mongo.url}") val connectionString: String,
        @Value("\${mongo.database}") val databaseName: String,
        @Value("\${telegram.lanxat.token}") val lanxatToken: String
    ) : CommandLineRunner {

    override fun run(vararg args: String?) {

        RegisterBots.registerBots(connectionString, databaseName, lanxatToken)
    }
}